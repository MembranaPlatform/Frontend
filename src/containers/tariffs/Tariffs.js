import React from 'react';
import { Col, Row, Container } from 'reactstrap';
import { FormattedMessage } from 'react-intl';

import Header from './components/Header';

class Tariffs extends React.PureComponent {
  state = {
    selectedTariff: this.props.auth.profile.billing.tariff,
  }

  data = {
    tariffs: ['free', 'premium', 'pro'],
    monthlyPrice: [{ tariff: 'free', value: '0' }, { tariff: 'premium', value: '1000 MBN ($10)' }, { tariff: 'pro', value: '2500 MBN ($25)' }],
    trading: [{ tariff: 'free', value: 'V' }, { tariff: 'premium', value: 'V' }, { tariff: 'pro', value: 'V' }],
    apiExchange: [{ tariff: 'free', value: '3' }, { tariff: 'premium', value: '25' }, { tariff: 'pro', value: 'NO LIMITS' }],
    orders: [{ tariff: 'free', value: '25' }, { tariff: 'premium', value: '100' }, { tariff: 'pro', value: 'NO LIMITS' }],
    proofOfTrade: [{ tariff: 'free', value: 'V' }, { tariff: 'premium', value: 'V' }, { tariff: 'pro', value: 'V' }],
    trustManagement: [{ tariff: 'free', value: 'X' }, { tariff: 'premium', value: 'UP TO $5000' }, { tariff: 'pro', value: 'NO LIMITS' }],
    telegramNotifications: [{ tariff: 'free', value: 'X' }, { tariff: 'premium', value: 'V' }, { tariff: 'pro', value: 'V' }],
    statusIcon: [{ tariff: 'free', value: 'X' }, { tariff: 'premium', value: 'X' }, { tariff: 'pro', value: 'V' }],
  }

  onSelectTariff = (tariff) => () => {
    this.setState({ selectedTariff: tariff });
  }

  render = () => (
    <Container className="tariffs__container">
      <Row>
        <Col></Col>
        {this.data.tariffs.map(tariff => (
          <Col>
            <Header
              label={tariff}
              active={tariff === this.state.selectedTariff}
              onClick={this.onSelectTariff(tariff)}
            />
          </Col>
        ))}
      </Row>
      <Row>
        <Col>MONTHLY PRICE</Col>
        {this.data.monthlyPrice.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>TRADING</Col>
        {this.data.trading.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>API exchange</Col>
        {this.data.apiExchange.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>ORDERS: STOP, OCO, NO-LOCK</Col>
        {this.data.orders.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>PROOF-OF-TRADE</Col>
        {this.data.proofOfTrade.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>TRUST MANAGEMENT</Col>
        {this.data.trustManagement.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>TELEGRAM NOTIFICATIONS</Col>
        {this.data.telegramNotifications.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <Row>
        <Col>STATUS ICON</Col>
        {this.data.statusIcon.map(({ tariff, value }) => (
          <Col className={tariff === this.state.selectedTariff && 'active'}>{value}</Col>
        ))}
      </Row>
      <div className="tariffs__container-description">
        Choose a service plan and click "BUY NOW" button.
        The payment is done from your active ERC-20 wallet or thought a direct transaction.
        Choose what is more convenient for you. The service plan fee is paid in MBN tokens,
        based on the market price available at coinmarketcap. After payment, the service plan is activated for 20 days.
      </div>
      <div className="tariffs__container-button-wrapper">
        <button onClick={() => {}} type="button" className="btn active">
          <FormattedMessage id="tariffs.buyNow" />
        </button>
      </div>
    </Container>
  );
}

export default Tariffs;
