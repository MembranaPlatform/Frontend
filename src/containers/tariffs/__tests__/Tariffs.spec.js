import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Tariffs from '../Tariffs';

describe('Tariffs component', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const props = {
    loggedIn: false,
    billing: { tariff: 'free' },
    tariffs: [
      { _id: 'free' },
      { _id: 'premium' },
      { _id: 'pro' },
    ],
    fetchTariffs: jest.fn(),
    createPaymentRequest: jest.fn(),
    history: {
      push: jest.fn(),
    },
  };

  const container = shallow(<Tariffs {...props} />);
  const instance = container.instance();

  it('renders Tariffs component', () => {
    expect(container).toMatchSnapshot();
  });

  describe('onSelectTariff()', () => {
    const setStateSpy = jest.spyOn(instance, 'setState');
    const tariff = 'premium';

    it('when loggedIn === false', () => {
      instance.onSelectTariff(tariff)();
      expect(setStateSpy).not.toHaveBeenCalled();
    });

    it('when loggedIn === true', () => {
      container.setProps({ loggedIn: true });
      instance.onSelectTariff(tariff)();
      expect(setStateSpy).toHaveBeenCalled();
    });
  });


  it('componentDidMount()', () => {
    instance.componentDidMount();

    expect(instance.props.fetchTariffs).toHaveBeenCalled();
    expect(instance.props.fetchTariffs).toHaveBeenCalledTimes(2);
  });
});
