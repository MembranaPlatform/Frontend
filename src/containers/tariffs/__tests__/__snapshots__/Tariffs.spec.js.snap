// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tariffs component renders Tariffs component 1`] = `
<Container
  className="tariffs__container"
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Header
        active={true}
        label="free"
        onClick={[Function]}
        selectable={false}
      />
    </Col>
    <Col
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Header
        active={false}
        label="premium"
        onClick={[Function]}
        selectable={false}
      />
    </Col>
    <Col
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Header
        active={false}
        label="pro"
        onClick={[Function]}
        selectable={false}
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      MONTHLY PRICE
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      undefined MBN ($undefined)
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      undefined MBN ($undefined)
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      undefined MBN ($undefined)
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      TRADING
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      API exchange
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      ORDERS: STOP, OCO, NO-LOCK
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      PROOF-OF-TRADE
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      ASSETS UNDER MANAGEMENT
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      UP TO $undefined
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      UP TO $undefined
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      UP TO $undefined
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      TELEGRAM NOTIFICATIONS
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-cross"
      />
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-cross"
      />
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-cross"
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      STATUS ICON
    </Col>
    <Col
      className="active"
      key="free"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-cross"
      />
    </Col>
    <Col
      className=""
      key="premium"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-cross"
      />
    </Col>
    <Col
      className=""
      key="pro"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span
        className="icon-checkmark"
      />
    </Col>
  </Row>
  <div
    className="tariffs__container-description"
  >
    Choose a service plan and click "BUY NOW" button. The payment can be done from your active ERC-20 wallet or thought a direct transaction. The service plan fee is paid in MBN tokens, based on the market price. After payment, the service plan is activated for 30 days.
  </div>
  <div
    className="tariffs__container-button-wrapper"
  >
    <button
      className="btn active"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        id="tariffs.buyNow"
        values={Object {}}
      />
    </button>
  </div>
  <ExpireInfo
    billing={
      Object {
        "tariff": "free",
      }
    }
  />
</Container>
`;
